File: HollowGoodsLibrary/src/main/java/com/hg/hollowgoods/Util/ExampleLoginUtils.java
package com.hg.hollowgoods.Util;

import com.hg.hollowgoods.Application.ExampleApplication;
import com.hg.hollowgoods.Bean.HGUser;
import com.hg.hollowgoods.DB.ExampleUserDBHelper;

/**
 * 登录工具类示例
 * Created by Hollow Goods on 2019-07-11.
 */
public class ExampleLoginUtils {

    public static boolean isLogin() {
        return ExampleApplication.createApplication().getUser() != null;
    }

    public static void initUser() {
        ExampleApplication.createApplication().setUser(new ExampleUserDBHelper().findFirst(HGUser.class));
    }

    public static void updateUser(HGUser user) {
        new ExampleUserDBHelper().updateAll(HGUser.class, user);
        ExampleApplication.createApplication().setUser(user);
    }

    public static HGUser getUser() {

        if (ExampleApplication.createApplication().getUser() != null) {
            return ExampleApplication.createApplication().getUser();
        }

        return new HGUser();
    }

    public static void destroyUser() {
        ExampleApplication.createApplication().setUser(null);
        new ExampleUserDBHelper().deleteAll(HGUser.class);
    }

}

File: HollowGoodsLibrary/src/main/java/com/hg/hollowgoods/Util/ExampleLoginUtils.kt
package com.hg.hollowgoods.Util

import com.hg.hollowgoods.Application.ExampleApplication
import com.hg.hollowgoods.Bean.HGUser
import com.hg.hollowgoods.DB.ExampleUserDBHelper

/**
 * 示例登录工具类
 * Created by HG on 2018-03-22.
 */
object ExampleLoginUtils {

    @JvmStatic
    fun isLogin(): Boolean {
        return ExampleApplication.createApplication().user != null
    }

    @JvmStatic
    fun initUser() {
        ExampleApplication.createApplication().user = ExampleUserDBHelper().find()
    }

    @JvmStatic
    fun updateUser(user: HGUser) {
        ExampleUserDBHelper().update(user)
        ExampleApplication.createApplication().user = user
    }

    @JvmStatic
    fun getUser(): HGUser? {

        return if (ExampleApplication.createApplication().user != null)
            ExampleApplication.createApplication().user
        else
            HGUser()
    }

    @JvmStatic
    fun destroyUser() {
        ExampleApplication.createApplication().user = null
        ExampleUserDBHelper().delete()
    }

}